### ISSUE TITLE
Implement support for negative divisor in the BigNumber and Fraction implementation of function `mod`

### ISSUE THREAD

**Issue by travellingprog** [josdejong/mathjs#3087]
**Describe the bug**

`gcd(x, y)` fails if `y` is negative and if `x` or `y` is a BigNumber. The `gcd(x, y)` function internally calls the `mod(x, y)` function. Unfortunately, the `mod(x, y)` function has been set to fail when a BigNumber is involved, if the divisor `y` is negative.

Potential solution would be for `gcd(x, y)` to internally call `mod(x, abs(y))` instead.

**To Reproduce**

```js
const { BigNumber, gcd } = require('mathjs');
gcd(-2, BigNumber(12)); // 2  -- No issue
gcd(BigNumber(12), -2); // Uncaught Error: Cannot calculate mod for a negative divisor
```

**Comment by josdejong** [comment]
Fixed via 8679c076176a8ea14be92985622dc4255800e527 (not yet published)

**Comment by josdejong** [comment]
Published now in `v12.1.0`
